### ISSUE TITLE
nearest not working when nb_cpu is set due to missing ray module

### ISSUE THREAD

**Issue by edg1983** [biocore-ntnu/pyranges#201]
Hi,

I have installed pyranges using conda as suggested on your repository and I want to use pyranges to annotate nearest intervals using the nearest method. It works fine with something like:
```
gr = gr1.nearest(gr2, strandness=False, suffix="_B", overlap=True)
``` 

However, when I try to use parallel processing with the nb_cpu option I get this error `ModuleNotFoundError: No module named 'ray'`
If I try to install ray using conda `conda install ray` the same error appears, while if I install ray using pip `pip install ray` I get a low of warnings from ray module and then it fails with KeyError for overlap key.

Any idea about this? Not essential for me to run in parallel right now, but good if this can be fixed.
Thanks!

**Comment by chartl-rancho** [comment]
Some more color on this:

The default install through pip installs ray in a functional way (though still warnings from way about using `pip install ray[default]` so that all dependencies are installed). However, the overlap key appears to drop stochastically.

On the first invocation of `gr.nearest(g2, overlap=True, strandedness='same', how=None, nb_cpu=4)` the method runs to completion. Upon the *second* invocation, a KeyError is raised for overlap. Could `ray.remote` be applied twice to the original function? It looks like it gets called with every invocation.

**Comment by chartl-rancho** [comment]
I also think this is the same as #200

**Comment by endrebak** [comment]
Hi and thanks for the question. I was on vacation. Will start to look at this now :)

Haven't used nb_cpu in a while so it might be that some of my code needs updating since `ray` is likely changing under our feet :)

**Comment by endrebak** [comment]
I should really have a warning about the fact that parallel only increases the speed when the data are *really* large.

**Comment by endrebak** [comment]
I cannot get Ray to work on my current MacBook so I cannot reproduce :/

nb_cpu is almost never faster on non-custom functions anyways since all the included functions in pyranges are so optimized that the overhead of using multiple cores makes everything slower anyways.

I should collect all ray stuff and create a new issue.

**Comment by worldveil** [comment]
@endrebak are you on an M1 laptop?
